addons/l10n_eg_edi_eta/models/account_edi_format.py

```diff
@@ -0,0 +1,426 @@
+            invoice.l10n_eg_eta_invoice_id.raw)['request']
+        if not invoice_json.get('signatures'):
+            return {invoice: {
+                'error':  _("Please make sure the invoice is signed"),
+                'blocking_level': 'info'
+            }
+            }
+        return self._l10n_eg_edi_post_invoice_web_service(invoice, invoice_json)
+
+    def _l10n_eg_edi_post_invoice_web_service(self, invoice, invoice_json):
+        token = self._l10n_eg_get_einvoice_token(invoice)
+        api_domain = self._l10n_eg_get_eta_api_domain()
+        if isinstance(token, dict) and token.get('error', False):
+            return {invoice: token}
+        request_url = '%s/api/v1.0/documentsubmissions' % api_domain
+
+        request_payload = {
+            'documents': [invoice_json]
+        }
+
+        data = json.dumps(request_payload, ensure_ascii=False, indent=4).encode('utf-8')
+
+        headers = {'Content-Type': 'application/json',
+                   'Authorization': 'Bearer %s' % token}
+        try:
+            request_response = requests.post(
+                request_url, data=data, headers=headers, timeout=(5, 10))
+            submission_data = request_response.json()
+        except (ValueError, requests.exceptions.ConnectionError, requests.exceptions.MissingSchema, requests.exceptions.Timeout, requests.exceptions.HTTPError) as ex:
+            return {invoice: {'error': str(ex), 'blocking_level': 'warning'}}
+
+        if submission_data:
+            if submission_data.get('error'):
+                return {invoice: {
+                    'error': submission_data.get('error'),
+                    'blocking_level': 'error'
+                }}
+            if submission_data.get('rejectedDocuments', False) and isinstance(submission_data.get('rejectedDocuments'), list):
+                return {invoice: {
+                    'error': str(submission_data.get('rejectedDocuments')[0].get('error')),
+                    'blocking_level': 'error'
+                }}
+            if (submission_data.get('submissionId') is not None
+                    and submission_data.get('acceptedDocuments')):
+                eta_invoice_json = json.loads(invoice.l10n_eg_eta_invoice_id.raw)
+                eta_invoice_json['response'] = {
+                    'l10n_eg_uuid': submission_data.get('acceptedDocuments')[0].get('uuid'),
+                    'l10n_eg_long_id': submission_data.get('acceptedDocuments')[0].get('longId'),
+                    'l10n_eg_internal_id': submission_data.get('acceptedDocuments')[0].get('internalId'),
+                    'l10n_eg_hash_key': submission_data.get('acceptedDocuments')[0].get('hashKey'),
+                    'l10n_eg_submission_id': submission_data.get('submissionId'),
+                }
+
+                invoice.l10n_eg_eta_invoice_id.raw = json.dumps(eta_invoice_json)
+                return {invoice: {}}
+
+            return {invoice: {
+                'success': False,
```

Comment:
It is a bit weird to return that without the appropriate error.  Is it possible we have no response json somehow?

Reply:
I don't think it is possible to not get a response json back but i had it in there for good measures